---
- hosts: localhost
  tasks:
    - name: Attempt and graceful roll back demo
      block:
        - debug:
            msg: 'I execute normally'
        - name: "i force a failure"
          command: /bin/false
          # loop:
          #   - hello
          #   - world
          # ignore_errors: yes
        - debug:
            msg: 'I never execute, due to the above task failing, :-('
      # rescue:
      #   - debug:
      #       msg: 'I caught an error'
      #   - name: i force a failure in middle of recovery! >:-)
      #     command: /bin/false
      #   - debug:
      #       msg: 'I also never execute :-('
      # always:
      #   - debug:
      #       msg: "This always executes"
      # ignore_errors: true